### PR TITLE
feat: update interface to support platforms with async URL lookups

### PR DESCRIPTION
**Pull request by defispartan**
Template for platforms to use if NFT details can't be extracted synchronously from URL

### CODE REVIEW

File: src/platform/nftPlatforms.ts [hunk 1]
```diff
@@ -82,11 +82,17 @@ export const NFT_PLATFORM_CONFIG: { [key: string]: NFTPlatform } = {
 };
 
 // Function to detect NFT details from URL
-export function detectNFTDetails(url: string): NFTExtraction | undefined {
+export async function detectNFTDetails(url: string): Promise<NFTExtraction | undefined> {
```
Comment (0xnogo) [inline src/platform/nftPlatforms.ts:91]: It would be great to update tests to keep the coverage of this function.

File: README.md [hunk 2]
```diff
@@ -112,11 +112,16 @@ type NFTPlatform = {
   platformName: string;
   platformLogoUrl: string;
   urlPattern: RegExp;
-  urlExtractor: (url: string) => NFTExtraction | undefined;
+  urlExtractor?: (url: string) => NFTExtraction | undefined;
```
Comment (0xnogo) [inline README.md]: Do we need to separate the async and sync logic? Could we have just the async logic and make sure we returned a resolved promise for the sync extractor?

File: src/platform/nftPlatforms.ts (hunk 1 above)
Comment (defispartan) [reply src/platform/nftPlatforms.ts:91]: Updated tests for async functions

File: README.md (hunk 2 above)
Comment (defispartan) [reply README.md]: Yeah you're right there isn't a benefit to separating so I removed the async function and updated urlExtractor to be compatible

**Review by 0xnogo — APPROVED** [review]
LGTM

**Pull request merged**